admin/class-admin-apple-news-list-table.php

```diff
@@ -252,19 +254,97 @@ public function prepare_items() {
 		$columns = $this->get_columns();
 		$this->_column_headers = array( $columns, array(), array() );
 
-		// Data fetch
+		// Build the default args for the query
 		$current_page = $this->get_pagenum();
-		$query = new WP_Query( apply_filters( 'apple_news_export_table_get_posts_args', array(
+		$args = array(
 			'post_type'     => $this->settings->get( 'post_types' ),
 			'post_status'	=> 'publish',
 			'posts_per_page' => $this->per_page,
 			'offset'         => ( $current_page - 1 ) * $this->per_page,
 			'orderby'        => 'ID',
 			'order'          => 'DESC',
-		) ) );
+		);
+
+		// Add the publish status filter if set
+		$publish_status = $this->get_publish_status_filter();
+		if ( ! empty( $publish_status ) ) {
+			switch ( $publish_status ) {
+				case 'published':
+					$args['meta_query'] = array(
+						'relation' => 'AND',
+						array(
+							'key' => 'apple_news_api_created_at',
```

Comment:
how unique is this meta key (and the others too)? If most/all posts have them, these meta queries can get to be _very_ slow